rs/migration_canister/tests/tests.rs

```diff
@@ -1092,7 +1008,75 @@ async fn parallel_migrations() {
             },
         )
         .await;
-        // TODO: should be Succeeded in the end.
-        assert!(matches!(status[0], MigrationStatus::InProgress { .. }));
+        let MigrationStatus::InProgress { ref status } = status[0] else {
+            panic!()
+        };
+        assert_eq!(status, "SourceDeleted");
+    }
+}
+
+#[tokio::test]
+async fn parallel_validations() {
+    const NUM_MIGRATIONS: usize = 550;
+    let Setup {
+        pic,
+        sources,
+        targets,
+        source_controllers,
+        ..
+    } = setup(Settings {
+        num_migrations: NUM_MIGRATIONS as u64,
+        ..Settings::default()
+    })
+    .await;
+    let sender = source_controllers[0];
+
+    for i in 0..45 {
+        migrate_canister(
+            &pic,
+            source_controllers[0],
+            &MigrateCanisterArgs {
+                source: sources[i],
+                target: targets[i],
+            },
+        )
+        .await
+        .unwrap();
+    }
+    // We have 5 slots left. Attempt to validate 505 calls.
+    // Of those, 5 should fail due to validation limit and 495 should fail due to
+    // rate limit.
+    let mut msg_ids = vec![];
+    for i in 45..NUM_MIGRATIONS {
+        let id = pic
+            .submit_call(
+                MIGRATION_CANISTER_ID.into(),
+                sender,
```

Comment:
It could be more interesting to have an invalid sender here so that the calls actually all fail due to validation and not due to the rate-limit of 50 requests per day: then we should actually see 500 invalid calls (that failed validations) and 5 rate-limited calls.